src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -283,8 +302,12 @@ public class PullRequestHandler : IHttpHandler
             // we won't get branchesList == null because of the success check
             for (int i = 0; i < branchesList.Length; i++)
             {
-                // Don't try to delete "master" branch, since we're checked out to it
-                if (!branchesList[i].EndsWith("master"))
+                // Don't try to delete "master" branch or detached since we're checked out to it
+                if (branchesList[i].Equals("master"))
```

Comment:
>if (branchesList[i].Equals("master")) [](start = 16, length = 37)

(Non-blocking) question only: the change from EndsWith to Equals is possible because of the Trim of "*", right?